src/migration/PoC_dfusion.js

```diff
@@ -11,48 +11,34 @@ async function migrate({ artifacts, deployer, network, account, web3, maxTokens
       web3,
     })
   }
-  const TokenOWLProxy = getDependency(
+  const feeToken = await getDeployedDependency(
     artifacts,
-    network,
     deployer,
     account,
     "@gnosis.pm/owl-token/build/contracts/TokenOWLProxy"
   )
-  const fee_token = await TokenOWLProxy.deployed()
 
-  const BatchExchange = getDependency(
+  const BatchExchange = getArtifact(artifacts, deployer, account, "@gnosis.pm/dex-contracts/build/contracts/BatchExchange")
+  const biMap = await getDeployedDependency(
     artifacts,
-    network,
-    deployer,
-    account,
-    "@gnosis.pm/dex-contracts/build/contracts/BatchExchange"
-  )
-  const BiMap = getDependency(
-    artifacts,
-    network,
     deployer,
     account,
     "@gnosis.pm/solidity-data-structures/build/contracts/IdToAddressBiMap"
   )
-  const IterableAppendOnlySet = getDependency(
+  const iterableAppendOnlySet = await getDeployedDependency(
     artifacts,
-    network,
     deployer,
     account,
     "@gnosis.pm/solidity-data-structures/build/contracts/IterableAppendOnlySet"
   )
 
-  // Hack to populate truffle artifact data correctly for linked libraries.
-  await BiMap.deployed()
-  await IterableAppendOnlySet.deployed()
-
   //linking libraries
-  await deployer.link(BiMap, BatchExchange)
-  await deployer.link(IterableAppendOnlySet, BatchExchange)
+  await deployer.link(biMap, BatchExchange)
```

Comment:
Does `BatchExchange.link(biMap)` do the same thing?

Reply:
During migrations, truffle expects you to do the linking with deployer, and for tests you would use your command.

But I am not sure about the inner workings of why it is like that